src/main/java/org/sonarlint/intellij/ui/traffic/light/SonarLintTrafficLightAction.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2023 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.ui.traffic.light
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.Presentation
+import com.intellij.openapi.actionSystem.ex.CustomComponentAction
+import com.intellij.openapi.editor.Editor
+import org.sonarlint.intellij.actions.SonarLintToolWindow
+import org.sonarlint.intellij.common.util.SonarLintUtils
+import javax.swing.JComponent
+
+class SonarLintTrafficLightAction(private val editor: Editor) : AnAction(), CustomComponentAction {
+
+    override fun createCustomComponent(presentation: Presentation, place: String): JComponent {
+        return SonarLintTrafficLightWidget(this, presentation, place, editor)
+    }
+
+    override fun update(e: AnActionEvent) {
+        val component = e.presentation.getClientProperty(CustomComponentAction.COMPONENT_KEY) as SonarLintTrafficLightWidget?
+        component?.refresh()
+    }
```

Comment:
I implemented that to have something quickly, but I'm not sure it's sufficient to refresh the view. How often is the update called? Shouldn't we listen for changes in the findings? If it's refreshed very often maybe it's sufficient

Reply:
It gets called very often (up to multiple times in seconds), and I think it is not necessary (I'm even afraid it has negative impacts on the resources), I will try to come up with a better solution 👍🏻